app/app.go

```diff
@@ -47,27 +65,44 @@ func NewApp(cfg *config.Config) *App {
 
 	signer := vote.NewVoteSigner(ethcommon.Hex2Bytes(cfg.VotePoolConfig.BlsPrivateKey))
 	voteDataHandler := vote.NewDataHandler(daoManager, executor)
-	voteProcessor := vote.NewVoteProcessor(cfg, daoManager, signer, executor, voteDataHandler)
+	voteCollector := vote.NewVoteCollector(cfg, daoManager, executor, voteDataHandler)
+	voteBroadcaster := vote.NewVoteBroadcaster(cfg, daoManager, signer, executor, voteDataHandler)
+	voteCollator := vote.NewVoteCollator(cfg, daoManager, signer, executor, voteDataHandler)
 
 	txDataHandler := submitter.NewDataHandler(daoManager, executor)
 	txSubmitter := submitter.NewTxSubmitter(cfg, executor, daoManager, txDataHandler)
 
 	return &App{
-		executor:      executor,
-		eventMonitor:  monitor,
-		hashVerifier:  hashVerifier,
-		voteProcessor: voteProcessor,
-		txSubmitter:   txSubmitter,
+		executor:        executor,
+		eventMonitor:    monitor,
+		hashVerifier:    hashVerifier,
+		voteCollector:   voteCollector,
+		voteBroadcaster: voteBroadcaster,
+		voteCollator:    voteCollator,
+		txSubmitter:     txSubmitter,
 	}
 }
 
 func (a *App) Start() {
 	go a.executor.UpdateAttestedChallengeIdLoop()
 	go a.executor.UpdateHeartbeatIntervalLoop()
 	go a.executor.CacheValidatorsLoop()
+	go a.executor.GetHeightLoop()
 	go a.eventMonitor.ListenEventLoop()
 	go a.hashVerifier.VerifyHashLoop()
-	go a.voteProcessor.SignVoteLoop()
-	go a.voteProcessor.CollectVotesLoop()
-	a.txSubmitter.SubmitTransactionLoop()
+	go a.voteCollector.CollectVotesLoop()
+	go a.voteBroadcaster.BroadcastVotesLoop()
+	go a.voteCollator.CollateVotesLoop()
+	go a.txSubmitter.SubmitTransactionLoop()
```

Comment:
The last loop should not run in a goroutine, otherwise the program will finish.

Reply:
fixed